### ISSUE TITLE
Builders for continuous integration

### ISSUE THREAD

**Issue by nathany** [fsnotify/fsnotify#136]
The combination of TravisCI (Linux/OS X) and AppVeyor (Windows) is not enough.

v1.3.0 was patched for linux/arm64, but we have no CI for arm64 to ensure that it keeps on working. We need a more comprehensive solution that covers arm64, ppc, and Solaris.

The best bet is to look into the Builders used by Go itself.
https://github.com/golang/go/wiki/DashboardBuilders

Perhaps one option would be to ask the Go Team if fsnotify could become a golang.org/x/fsnotify subrepository, but maybe that isn't necessary.

Of note, there still could be [API changes](https://github.com/fsnotify/fsnotify/issues?q=is%3Aopen+is%3Aissue+label%3AAPI) that warrant a v2.0 of fsnotify.

**Comment by nathany** [comment]
As far as local testing, I currently use https://github.com/nathany/vagrant-gopher for BSD and Linux, which I run on a Mac along with a Windows 10 VM in VMware Fusion. Solaris could probably be added, but that still only covers x86.

I recall some plans to make it possible to utilize the builders from a local computer while debugging issues, but I don't know where that is currently at.

**Comment by nathany** [comment]
From @bradfitz:

> Currently our builders can only run stuff in golang.org/x* and hosted on Gerrit.
> That's not fundamental, but it's just how things are currently implemented.

I will compose a message to golang-dev tonight after work to see where the community wants to take this.

**Comment by nathany** [comment]
There is also a matter of where the issue tracking is done in the future. In a subrepo or the Go language repo. The change from pull requests to CLs.

I'm fairly happy with GitHub's recent improvements to pull requests and my own workflows using hub, so it's really just the builders that we need to get going on.

**Comment by bradfitz** [comment]
I've also been working to make Github pull requests auto-convert into Gerrit reviews. (https://github.com/LetsUseGerrit/)

**Comment by nathany** [comment]
Posted to golang-dev to include other Go contributors in the discussion. https://groups.google.com/forum/#!topic/golang-dev/SVrNHQU1oEM

**Comment by nathany** [comment]
Opened an issue over in the Go repository. https://github.com/golang/go/issues/17312

**Comment by anthonyfok** [comment]
Hi Nathan,

Ubuntu runs an autopkgtest CI, with arm64 included, here:

* https://autopkgtest.ubuntu.com/packages/golang-fsnotify
* https://autopkgtest.ubuntu.com/packages/golang-fsnotify/cosmic/arm64

Granted, it is run only after when we upload a new Debian package for golang-fsnotify, but I think it does help, I think, because it reveals that one of the tests have always failed on arm64!

```
=== RUN   TestInotifyOverflow
--- FAIL: TestInotifyOverflow (58.47s)
	inotify_test.go:422: Not done
```

(I will file a separate issue about TestInotifyOverflow.)
